### ISSUE TITLE
Translation of smoker[0..1]?

### ISSUE THREAD

**Issue by noezarwin** [escribis/CyberResidencesOCL#2]
What does it mean?
How do we translate it in USE OCL?

**Comment by xcriptus** [comment]
## Rules for UML
- `smoker : boolean[0..1]` means that smoker is an optional attribute.
- `age : integer[0..1]` means that age is optional
- `age : integer[1]`   (or simply `age:integer`) means that age is mandatory, so it is really a integer value that must be given. 

With [0..1], the attribute is possibly "undefined (null in databases).

## Rules for USE OCL

In principle it should be possible to declare optional attributes with OCL and use `age->isEmpty()` to test it. However, in **USE OCL all attributes are optional** (even not declared as such). So `smoker : boolean[0..1]` translates in `smoker : Boolean` which still means that smoker is optional in USE OCL.

In fact, mandatory attributes should be translate with a constraint. For example

```
    age : integer
```

should be translated into the attribute but also a constraint as following:

```
    age : Integer
    ...
    context Person: inv ageMandatory:
           self.age->isDefined
```

In practice this could be acheived easily only in the context of code generation.

## Summary

Summing up, the translation of `smoker` and `age` are analogous. You can assume that age is always defined, but you must use `isDefined` when using `smoker`.

Hope this helps.
